From 32cadff6da997419191a855933d21010619ec527 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 1 Jul 2024 09:00:13 +0200
Subject: [PATCH 1252/1695] drivers: stm32_rstctrl: fix unpaged resource

Fixes missing declaration of STM32MP1 reset controller compat data
as unpaged resource since it is used  by stm32mp_rcc_reset_id_to_rstctrl()
function called by fastcall service psci_system_reset() that requires
to resized in the unpaged segment on STM32MP15 variant where pager
is enabled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b8082b349eba455de166cbb999bf538c29636ea6 drivers: stm32_rstctrl: fix unpaged resource]
Change-Id: I78aa022a175a3aec9db9cb8e1d8f1d7ea8f6f908
[etienne: cherry-picked from 82d8cbbd8848708f37c7f43e6a965291c26fbfd4]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/390537
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/rstctrl/stm32mp1_rstctrl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/drivers/rstctrl/stm32mp1_rstctrl.c b/core/drivers/rstctrl/stm32mp1_rstctrl.c
index 2119d1f9f..c8c5d9f22 100644
--- a/core/drivers/rstctrl/stm32mp1_rstctrl.c
+++ b/core/drivers/rstctrl/stm32mp1_rstctrl.c
@@ -148,6 +148,7 @@ static struct rstctrl_ops *stm32_reset_get_ops(unsigned int id __unused)
 static const struct stm32_reset_data stm32mp1_reset_data = {
 	.get_rstctrl_ops = stm32_reset_get_ops
 };
+DECLARE_KEEP_PAGER(stm32mp1_reset_data);
 
 static const struct dt_device_match stm32_rstctrl_match_table[] = {
 	{
-- 
2.47.2

